pybamm/expression_tree/broadcasts.py

```diff
@@ -33,72 +32,84 @@ def __init__(
         self,
         child,
         broadcast_domain,
-        auxiliary_domains=None,
+        broadcast_auxiliary_domains=None,
         broadcast_type="full",
         name=None,
     ):
         # Convert child to scalar if it is a number
         if isinstance(child, numbers.Number):
             child = pybamm.Scalar(child)
+        # Convert domain to list if it's a string
+        if isinstance(broadcast_domain, str):
+            broadcast_domain = [broadcast_domain]
 
         if name is None:
             name = "broadcast"
 
         # perform some basic checks and set attributes
-        domain = self.check_and_set_domain_and_broadcast_type(
-            child, broadcast_domain, broadcast_type
+        domain, auxiliary_domains = self.check_and_set_domains(
+            child, broadcast_type, broadcast_domain, broadcast_auxiliary_domains
         )
         self.broadcast_type = broadcast_type
         self.broadcast_domain = broadcast_domain
-        if auxiliary_domains is None:
-            if child.domain != []:
-                auxiliary_domains = {"secondary": child.domain}
-            else:
-                auxiliary_domains = {}
         super().__init__(name, child, domain, auxiliary_domains)
 
-    def check_and_set_domain_and_broadcast_type(
-        self, child, broadcast_domain, broadcast_type
+    def _unary_simplify(self, simplified_child):
+        """ See :meth:`pybamm.UnaryOperator.simplify()`. """
+        return self._unary_new_copy(simplified_child)
+
+
+class PrimaryBroadcast(Broadcast):
+    """A node in the expression tree representing a primary broadcasting operator.
+    Broadcasts in a `primary` dimension only. That is, makes explicit copies
```

Comment:
I'm not sure the "That is, makes explicit copies" is particularly clear?

Reply:
oh yeah I got distracted while writing this 😅